### PR TITLE
Documentation of Protocol structs

### PR DESCRIPTION
**Pull request by SmartMonkey-git**

### CODE REVIEW

File: tycho-indexer/src/extractor/evm/mod.rs [hunk 1]
```diff
@@ -426,6 +426,25 @@ impl TvlChange {
         })
     }
 }
+
+/// Represents the static parts of a protocol component.
+///
+/// `ProtocolComponent` provides detailed descriptions of the functionalities a protocol
+/// such as swap pools that enables the exchange of two tokens.
```
Comment (dianacarvalho1) [inline tycho-indexer/src/extractor/evm/mod.rs]:
```suggestion
/// `ProtocolComponent` provides detailed descriptions of the functionalities a protocol,
/// for example, swap pools that enables the exchange of two tokens.
```

File: tycho-indexer/src/extractor/evm/mod.rs [hunk 2]
```diff
@@ -426,6 +426,25 @@ impl TvlChange {
...
+///
+/// A `ProtocolComponent` can be associated with an `Account`, and it has an identifier (`id`) that
+/// can be either the on-chain address or a custom one. It belongs to a specific `ProtocolSystem`
+/// and has a `ProtocolType`declaring its behaviour e.g., swap, lend, bridge. The component is
```
Comment (dianacarvalho1) [inline tycho-indexer/src/extractor/evm/mod.rs]:
```suggestion
/// can be either the on-chain address or a custom one. It belongs to a specific `ProtocolSystem`
/// and has a `ProtocolTypeID` that associates it with a `ProtocolType` that describes its behaviour e.g., swap, lend, bridge. The component is
```

File: tycho-indexer/src/extractor/evm/mod.rs [hunk 3]
```diff
@@ -426,6 +426,25 @@ impl TvlChange {
...
+/// associated with a specific `Chain` and holds information about tradable tokens, related contract
+/// IDs, and static attributes.
+///
+/// A `ProtocolComponent` can have a one-to-one or many-to-one relationship with components.
```
Comment (dianacarvalho1) [inline tycho-indexer/src/extractor/evm/mod.rs]: I think you mean contracts, no components here

File: tycho-indexer/src/extractor/evm/mod.rs [hunk 4]
```diff
@@ -426,6 +426,25 @@ impl TvlChange {
...
+/// For example, `UniswapV2` and `UniswapV3` have a one-to-one relationship one component (pool) one
+/// contract, while `Ambient` has a many-to-one relationship with a single component and multiple
+/// contracts.
```
Comment (dianacarvalho1) [inline tycho-indexer/src/extractor/evm/mod.rs]: one-to-many no?

Comment (dianacarvalho1) [reply tycho-indexer/src/extractor/evm/mod.rs]: wait this raises a question for me... then when we have a new ambient pool we don't create a new ProtocolComponent object??? we need to update the existing one? 🤔

File: tycho-indexer/src/models.rs [hunk 5]
```diff
@@ -19,6 +19,7 @@ pub enum Chain {
     ZkSync,
 }
 
+/// Represents the ecosystem to which a `ProtocolComponent` belongs.
```
Comment (dianacarvalho1) [inline tycho-indexer/src/models.rs:22]: ecosystem?

Comment (SmartMonkey-git) [reply tycho-indexer/src/models.rs:22]: Yes, uniswap uses the word to describe the entirety of their contracts and those who integrate/interact with them.
https://docs.uniswap.org/contracts/v2/concepts/protocol-overview/ecosystem-participants

I have also read it on confluence planning. Like [here](https://datarevenue.atlassian.net/wiki/spaces/DEFI/pages/2310438916/Token+Plan).

> List of the preliminary use-cases of the token and ecosystem components.

File: tycho-indexer/src/extractor/evm/mod.rs (hunk 4 above)
Comment (SmartMonkey-git) [reply tycho-indexer/src/extractor/evm/mod.rs]: I would think so, yes.

Comment (dianacarvalho1) [reply tycho-indexer/src/extractor/evm/mod.rs]: that is not what we have been doing 😐

File: tycho-indexer/src/extractor/evm/mod.rs (hunk 3 above)
Comment (dianacarvalho1) [reply tycho-indexer/src/extractor/evm/mod.rs]: this was not addressed

File: tycho-indexer/src/extractor/evm/mod.rs (hunk 4 above)
Comment (dianacarvalho1) [reply tycho-indexer/src/extractor/evm/mod.rs]: > one-to-many no?

this was not addressed either

**Pull request merged**